generic/openblas.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Pooja B Surya <pooja@linux.vnet.ibm.com>
+
+import os
+
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Openblas(Test):
+
+    def setUp(self):
+        smm = SoftwareManager()
+        for package in ['libopenblas-base', 'libopenblas-dev', 'git',
+                        'gfortran']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.error(package + ' is needed for the test to be run')
```

Comment:
skip test is more appropriate here as if package not available test should skip please use `self.skip() ` instead of `self.error` ?